doc/VTGateV3Features.md

```diff
@@ -0,0 +1,86 @@
+* V2: In this version, the keyspace id was required instead of the shard. This allowed the app to be agnostic of the number of shards.
+
+With V3, the app does not need to specify any routing info. It just sends the query to VTGate as if it's a single database. Apart from simplifying the API, there are some additional benefits:
+* Database compliant drivers can be built for each client language. This will also allow for integration with third party tools that work with such drivers.
+* V3 can aspire to satisfy the full SQL syntax. This means that it will be able to perform cross-shard joins, aggregations and sorting.
+* Easier migration: an application that was written to use a single database can be trivially changed to use Vitess, and then the database can be scaled from underneath without changing much of the app.
+
+## Feature set
+The [V3 design](https://github.com/youtube/vitess/blob/master/doc/VTGateV3.md) is quite elaborate. If necessary, it will allow you to plug in custom indexes and sharding schemes. However, it comes equipped with some pre-cooked recipes that satisfy the immediate needs of the real-world:
+
+### Knowing where tables are
+As your database grows, you will not only be sharding it, you will also be splitting it vertically by migrating tables from one database to another. V3 will be able to keep track of this. The app will only have to refer a table by name, and VTGate will figure out how to route the query to the correct database.
+
+The vitess workflow also ensures that such migrations are done transparently with virtually no downtime.
+
+### Sharding schemes
+At its core, vitess uses range-based sharding, where the sharding column is typically a number or a varbinary. However, allowing data to be accessed only by the sharding key limits the flexibility of an application. V3 comes with a set of new indexing schemes that are built on top of range-based sharding.
+
+#### Basic sharding key
+If the application already has a well-distributed sharding key, you just have to tell VTGate what those keys are for each table. VTGate will correctly route your queries based on input values or the WHERE clause.
+
+#### Hashed sharding key
+If the application's sharding key is a monotonically increasing number, then you may not get well-balanced shards. In such cases, you can ask V3 to route queries based on the hash of the main sharding key.
+
+Vitess's filtered replication currently requires that the hash value be physically present as a column in each table. To satisfy this need, you still need to create a column to store this hash value. However, V3 will take care of populating this on your behalf.
+
+*This restriction will soon be removed once filtered replication is changed to also perform the same hashing.*
+
+#### Auto-increment columns
```

Comment:
I think these should be listed here, because this is a features doc. If we were to build a user guide, these topics would belong to the advanced section. Hopefully, if the UI is intuitive enough, we can avoid building the user guide.